### ISSUE TITLE
TriMesh::write without texture coordinates

### ISSUE THREAD

**Issue by gaborpapp** [cinder/Cinder#773]
`TriMesh::write` fires an assert in glNext if the `TriMesh` has no texture coords in https://github.com/cinder/Cinder/blob/glNext/src/cinder/TriMesh.cpp#L460
although texcoords should not be required.

**Comment by richardeakin** [comment]
Looks like TriMesh read() / write() aren't as flexible as glNext's TriMesh yet. What if instead of storing only the number elements, we also store the dims? Ex, the first 8 numbers in the file would be:

numPos, sizePos, numNormals, sizeNormals, numTexCoords0, sizeTexCoord0.

I don't see why we shouldn't read and write texCoords1-3 as well..

Then we'd be able to remove those asserts.

**Comment by gaborpapp** [comment]
Good suggestion. It would be nice to support at least as much as ObjLoader, so read obj's can be written with TriMesh::write() easily.

**Comment by richardeakin** [comment]
In addition, I propose we add an optional argument to the write() method that will enable writing tangents / bitangents:

```
void	write( DataTargetRef out, bool writeTangents = false ) const;
```

It seems that if the usefulness of this method is to optimize load times, users could benefit from not having to calculate tangents / bitangents again a the cost of extra memory.

**Comment by richardeakin** [comment]
I've begun work on this in the following branch: https://github.com/richardeakin/Cinder/tree/glNext_trimesh_io

Still need to add some test code and optional writeTangents.

**Comment by gaborpapp** [comment]
Great, thanks!

**Comment by richardeakin** [comment]
Fixed in #832.
